### ISSUE TITLE
novnvc for high latency connections

### ISSUE THREAD

**Issue by shivaenigma** [kanaka/noVNC#563]
I did an interactive user session, in which  there is a frame update for every mouse button scroll

Network speed: 100Mbps
Latency : 10ms
Tightvnc viewer frame rate: 13.19 fps
noVnc viewer frame rate:  8.39 fips

Is there anyway I can get a higher rate frame. I saw https://github.com/kanaka/noVNC/issues/221 , but looks like it is no longer applicable on the current source. Note: I don't mind little bit high CPU / network usage for high latency connection

Not sure how tight vnc has implemented this to get higher frame rate. Quoting  http://www.turbovnc.org/pmwiki/uploads/About/vglperf21.pdf 
```
Beginning with TurboVNC 0.3.2, an attempt was made to pipeline at least part of the RFB protocol.
This was accomplished by modifying the TurboVNC viewer such that it sent a new framebuffer update
request before decoding the previous update, rather than after.  This allowed at least part of the network
latency   to   be   hidden,   which   significantly   improved   TurboVNC's   performance   on   high-latency
networks.
```
Is this already implemented. Can I do anything to get a high frame rate ?

**Comment by shivaenigma** [comment]
Hi,
Just wanted to know, is this something that can be implemented in noVnc , we would be happy to help in development and testing. Just wanted to know the feasibility  of getting  this in

**Comment by DirectXMan12** [comment]
Sorry, I've been a bit busy lately, and I wanted to get a chance to look into what you were talking about before responding.  We'd certainly be glad to accept pull requests that improved performance, especially if you had some numbers to back up the change.

**Comment by samhed** [comment]
What is described in that report is called continuous updates in the RFB protocol. This is definitely possible to implement and is something I would be very glad to help out with! Unfortunately I am a bit limited on time available for this, I personally won't be able to start this project in the near-time.

Anyway, other than asking the VNC server to enable this feature, noVNC will also have to support the Fence extension. Information is available here:

https://github.com/rfbproto/rfbproto/blob/master/rfbproto.rst#continuousupdates-pseudo-encoding
https://github.com/rfbproto/rfbproto/blob/master/rfbproto.rst#fence-pseudo-encoding

https://github.com/rfbproto/rfbproto/blob/master/rfbproto.rst#enablecontinuousupdates
https://github.com/rfbproto/rfbproto/blob/master/rfbproto.rst#endofcontinuousupdates

https://github.com/rfbproto/rfbproto/blob/master/rfbproto.rst#clientfence
https://github.com/rfbproto/rfbproto/blob/master/rfbproto.rst#serverfence

**Comment by astrand** [comment]
https://github.com/kanaka/noVNC/pull/623

**Comment by samhed** [comment]
Closed by #623
